game-app/game-headed/src/main/java/games/strategy/triplea/ui/menubar/ViewMenu.java

```diff
@@ -52,486 +52,488 @@
+                            panel.add(spinner, BorderLayout.CENTER);
+                            final JPanel buttons = new JPanel();
+                            final JButton fitWidth = new JButton("Fit Width");
+                            buttons.add(fitWidth);
+                            final JButton fitHeight = new JButton("Fit Height");
+                            buttons.add(fitHeight);
+                            final JButton reset = new JButton("Reset");
+                            buttons.add(reset);
+                            panel.add(buttons, BorderLayout.SOUTH);
+                            fitWidth.addActionListener(
+                                    event -> {
+                                        final double screenWidth = frame.getMapPanel().getWidth();
+                                        final double mapWidth = frame.getMapPanel().getImageWidth();
+                                        double ratio = screenWidth / mapWidth;
+                                        ratio = Math.max(frame.getMapPanel().getMinScale(), ratio);
+                                        ratio = Math.min(1, ratio);
+                                        model.setValue((int) Math.round(ratio * 100));
+                                    });
+                            fitHeight.addActionListener(
+                                    event -> {
+                                        final double screenHeight = frame.getMapPanel().getHeight();
+                                        final double mapHeight = frame.getMapPanel().getImageHeight();
+                                        double ratio = screenHeight / mapHeight;
+                                        ratio = Math.max(frame.getMapPanel().getMinScale(), ratio);
+                                        model.setValue((int) Math.round(ratio * 100));
+                                    });
+                            reset.addActionListener(event -> model.setValue(100));
+                            final int result =
+                                    JOptionPane.showOptionDialog(
+                                            frame,
+                                            panel,
+                                            "Choose Map Zoom",
+                                            JOptionPane.OK_CANCEL_OPTION,
+                                            JOptionPane.PLAIN_MESSAGE,
+                                            null,
+                                            new String[] {"OK", "Cancel"},
+                                            0);
+                            if (result != 0) {
+                                return;
+                            }
+                            final Number value = (Number) model.getValue();
+                            frame.getMapPanel().setScale(value.doubleValue() / 100);
+
+                            frame.getGame().getData().notifyMapZoomChanged(value.intValue());
```

Comment:
WDYT of setting the value directly on `bottomBar`?

EG, something like:

```
              frame.getBottomBar().setMapZoom(value);
```

I see this as advantageous for two reasons:
-  less 'plumbing' 
-  no need to route a UI concern through `GameData`